src/directives/column-reorder.js

```diff
@@ -0,0 +1,223 @@
+                        } else if (offset > 0) {
+                            // Insert After
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex + 1, 0, oldColumn);
+                        } else {
+                            // Swap Column Places
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            var newColumn = rowTemplate.splice(newColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                            rowTemplate.splice(oldColIndex, 0, newColumn);
+                        }
+
+                        var newRowTemplate = '';
+                        for(var i = 0; i < rowTemplate.length; i++) {
+                            newRowTemplate += rowTemplate[i].outerHTML;
+                        }
+
+                        templateMapper[rowId] = newRowTemplate;
+                    });
+
+                    $scope.$broadcast('drawTableRows');
+                }
+
+            },
+            compile: function($element, $attrs) {
+                var headerRow = $element[0].querySelector('thead tr');
+                headerRow.setAttribute('reorderable-column-template-mapper', getId());
+
+                var headers = headerRow.querySelectorAll('th');
+                var headerCount = headers.length;
+                while (headerCount--) {
+                    headers[headerCount].setAttribute('reorderable-header-drag-drop', '');
+                }
+
+                var bodyRows = $element[0].querySelectorAll('tbody tr');
+                var bodyRowCount = bodyRows.length;
+                while (bodyRowCount--) {
+                    bodyRows[bodyRowCount].setAttribute('reorderable-column-template-mapper', getId());
+                }
+
+            }
+        }
+    });
+
+    module.directive('reorderableColumnTemplateMapper', function($compile) {
+        return {
+            require: '^reorderableColumns',
+            compile: function($element, $attrs) {
+                var directiveName = this.name;
+                var rowId = $attrs[directiveName];
+
+                var uncompiledHtml = $element[0].innerHTML;
+                $element.empty();
+
+                return {
+                    pre: function (scope, element, attrs, ReorderColCtrl) {
+                        ReorderColCtrl.setTemplate(rowId, uncompiledHtml);
+
+                        var buildRow = function() {
+                            element.empty();
+                            var rowTemplate = ReorderColCtrl.getTemplate(rowId);
+                            rowTemplate = angular.element(rowTemplate);
+                            element.append(rowTemplate);
+                            $compile(rowTemplate)(scope);
+                        };
+                        buildRow();
+
+                        scope.$on('drawTableRows', buildRow);
```

Comment:
The example page is having this called 2060 times..... ouch.  Performance killer (and it is noticeable).  Do we need to build the row if we aren't moving it?